hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/dto/AttributeSearchRequestTest.kt

```diff
@@ -0,0 +1,616 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services.dto
+
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.jsonMapper
+import com.fasterxml.jackson.module.kotlin.kotlinModule
+import com.fasterxml.jackson.module.kotlin.readValue
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import java.time.LocalDate
+
+class AttributeSearchRequestTest {
+
+  private val objectMapper = jsonMapper { addModules(kotlinModule(), JavaTimeModule() )}
+
+  @Test
+  fun `firstName is John`() {
+   val request = objectMapper.readValue<AttributeSearchRequest>("""{
+            "matchers": [
+              {
+                "joinType": "AND",
+                "textMatchers": [
+                  {
+                    "attribute": "firstName",
+                    "condition": "IS",
+                    "searchTerm": "John"
+                  }
+                ]
+              }
+            ]
+          }""".trimIndent())
```

Comment:
Best to read the test name and the JSON to perform the search. The assertions afterwards just prove that the JSON is parsed into our query objects correctly.

With a single value the `joinType` is kind of irrelevant but didn't want to default it to AND. Maybe we should?